compiler/modules/front/const.bal

```diff
@@ -517,22 +520,46 @@ function foldVarRefExpr(FoldContext cx, t:SemType? expectedType, s:VarRefExpr ex
     }
 }
 
-function foldFloatLiteralExpr(FoldContext cx, t:SemType? expectedType, s:FpLiteralExpr expr) returns s:ConstValueExpr|FoldError {
-    // This will need to change when we support decimal
-    float|error result = floatFromDecimalLiteral(expr.untypedLiteral);
-    if result is float {
+function foldFpLiteralExpr(FoldContext cx, t:SemType? expectedType, s:FpLiteralExpr expr) returns s:ConstValueExpr|FoldError {
+    float|decimal|error result;
+    string ty;
+    s:FpTypeSuffix? typeSuffix = expr.typeSuffix;
+    string untypedLiteral = expr.untypedLiteral;
+    if typeSuffix is s:DECIMAL_TYPE_SUFFIX {
+        result = decimalFromFpLiteral(untypedLiteral);
+        ty = "decimal";
+    }
+    else if typeSuffix is s:FLOAT_TYPE_SUFFIX {
+        result = floatFromFpLiteral(untypedLiteral);
+        ty = "float";
+    }
+    else {
+        if expectedType == () || t:isSubtypeSimple(expectedType, t:FLOAT) {
+            result = floatFromFpLiteral(untypedLiteral);
+            ty = "float";
+        }
+        else if t:isSubtypeSimple(expectedType, t:DECIMAL) {
+            result = decimalFromFpLiteral(untypedLiteral);
+            ty = "decimal";
+        }
+        else {
+            result = floatFromFpLiteral(untypedLiteral);
+            ty = "float";            
+        }
+    }
+    if result is float|decimal {
         return { startPos: expr.startPos, endPos: expr.endPos, value: result };
     }
     else {
-        return cx.semanticErr("invalid float literal", cause=result, pos=expr.startPos);
+        return cx.semanticErr("invalid " + ty + " literal", cause=result, pos=expr.startPos);
```

Comment:
Do we ever each this line, if so for what input? if not we should simplify the code by removing error message.

Reply:
Removed